pkg/query/endpointset.go

```diff
@@ -228,7 +238,17 @@ func (c *endpointSetNodeCollector) Collect(ch chan<- prometheus.Metric) {
 			if storeType != nil {
 				storeTypeStr = storeType.String()
 			}
-			ch <- prometheus.MustNewConstMetric(c.connectionsDesc, prometheus.GaugeValue, float64(occurrences), externalLabels, storeTypeStr)
+			// select only required labels
+			lbls := []string{}
+			for _, lbl := range c.requiredLabels {
```

Comment:
I wonder if we can build this upfront. `c.requiredLabels` are never changing once `newEndpointSet...` constructor is done, right?

This code looks like they are changing which gives additional cognitive load for readers of this code - as is "oh, it's changing between collections, I wonder where...."

Reply:
yh this could be done but would take a bit more refactoring, I think because the actual label values are not known until when collecting or updating.